ui/vendor/_directional.scss

```diff
@@ -0,0 +1,63 @@
+// directional-scss | Author: Tyson Matanich (http://matanich.com), 2013 | License: MIT
+$dir: ltr !default;
+
+// Default $dir if not valid
+@if $dir != ltr and $dir != rtl {
+    $dir: ltr;
+}
+
+@function if-ltr($if, $else: null) {
+  @if $dir != rtl {
+    @return $if;
+  }
+  @else {
+    @return $else;
+  }
+}
+
+@function if-rtl($if, $else: null) {
+  @return if-ltr($else, $if);
+}
+
+$left: if-ltr(left, right);
+$right: if-ltr(right, left);
```

Comment:
It is confusing that right can mean left and left can mean right, the value of right and left should be absolute to avoid confusion. 

If we want to be rtl friendly we the names should be generalized, like `start` and `end`. 

```html
<div class="slds-rtl"> 
    <p class="slds-text-align--left">I should stay on the left no matter what.</p>
    <p class="slds-text-align--start">
        I should go right when a parent has slds-rtl, or left otherwise.
    </p>

      <p class="slds-text-align--end">
         I should go left when a parent has slds-rtl, or left otherwise.
      </p>
 
      <div class="slds-ltr">
         <p class="slds-text-align--start">
             I am going left coz my closest parent has `slds-ltr`
         </p>
      </div>
</div>
```